src/webui/abstractwebapplication.cpp

```diff
@@ -307,10 +308,12 @@ void AbstractWebApplication::increaseFailedAttempts()
 
 bool AbstractWebApplication::isAuthNeeded()
 {
-    return (env_.clientAddress != QHostAddress::LocalHost
-            && env_.clientAddress != QHostAddress::LocalHostIPv6
-            && env_.clientAddress != QHostAddress("::ffff:127.0.0.1"))
-            || Preferences::instance()->isWebUiLocalAuthEnabled();
+    qDebug("Checking auth rules against client address %s", qPrintable(env_.clientAddress.toString()));
+    if (Utils::Net::isLANAddress(env_.clientAddress))
```

Comment:
> so adding that additional check would break the feature for some client environments. 

As well as now it is unusable for others. For instance, your guesthouse gives IPs from sub-net 10.0.0.0/24. You set a router in your room and assign to your laptop, phone and tablet IPs from 192.168.0.0/16. But you can't disable authentication for LAN clients, because all the guesthouse will be able to use your qBt instance.